cogeo_mosaic/backends/base.py

```diff
@@ -136,19 +147,59 @@ def update(
 
     def assets_for_tile(self, x: int, y: int, z: int) -> List[str]:
         """Retrieve assets for tile."""
-        return self.get_assets(x, y, z)
+        mosaic_tms = self.mosaic_def.tilematrixset or WEB_MERCATOR_TMS
+        if self.tms.rasterio_crs == mosaic_tms.rasterio_crs:
+            return self.get_assets(x, y, z)
+
+        xmin, ymin, xmax, ymax = self.tms.bounds(x, y, z)
+        return self.assets_for_bbox(
+            xmin,
+            ymin,
+            xmax,
+            ymax,
+            coord_crs=self.tms.rasterio_geographic_crs,
+        )
 
-    def assets_for_point(self, lng: float, lat: float) -> List[str]:
+    def assets_for_point(
+        self,
+        lng: float,
+        lat: float,
+        coord_crs: CRS = WGS84_CRS,
+    ) -> List[str]:
         """Retrieve assets for point."""
-        tile = self.tms.tile(lng, lat, self.quadkey_zoom)
+        mosaic_tms = self.mosaic_def.tilematrixset or WEB_MERCATOR_TMS
+        if coord_crs != mosaic_tms.rasterio_geographic_crs:
```

Comment:
aren't all long/lats always geographic coordinates? Maybe the check here could allow for projected coords also (northing/easting/etc).

Reply:
good question. 

in rio-tiler we always assume the coordinate to be geographic and in X, Y order 🤷. 

my understanding is that user should always provide coordinates in X, Y order

Reply:
okay forgot that we want to allow for other coord_crss here so lgtm